tests/consistent.rs

```diff
@@ -0,0 +1,213 @@
+pub fn backends_are_consistent(re: &str) -> Result<u64, String> {
+    let standard_backends = vec![
+        ("bounded_backtracking_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_re",
+            try!(ExecBuilder::new(re)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    let utf8bytes_backends = vec![
+        ("bounded_backtracking_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    let bytes_backends = vec![
+        ("bounded_backtracking_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    Ok(try!(string_checker::check_backends(&standard_backends))
+       + try!(string_checker::check_backends(&utf8bytes_backends))
+       + try!(bytes_checker::check_backends(&bytes_backends)))
+}
+
+//
+// A consistency checker parameterized by the input type (&str or &[u8]).
+//
+
+macro_rules! checker {
+    ($module_name:ident, $regex_type:path, $mk_input:expr) => {
+        mod $module_name {
```

Comment:
I'd recommend out-denting the module declaration, and then indenting its contents from there.